fineract-provider/src/main/java/org/apache/fineract/portfolio/client/service/ClientReadPlatformServiceImpl.java

```diff
@@ -258,6 +259,13 @@ private String buildSqlStringFromClientCriteria(String schemaSql, final SearchPa
             extraCriteria += " and c.display_name like ? ";
         }
 
+        if (status != null) {
+            ClientStatus clientStatus = ClientStatus.fromString(status);
+            if (clientStatus.getValue() != 0) {
```

Comment:
So this would mean you can't search for clients with status = INVALID. Is this correct behaviour - shouldn't we still allow searching even if the value you pass in is INVALID? I.e. to find any INVALID entries (if any)...

At the same time it would be good to validate the provided input string to ensure it has actually mapped to a valid value, and throw PlatformValidationException if it doesn't. 

So you should be able to search for any invalid entries by sending in clientStatusType.invalid - but if you send in clientStatusType.blah then you should get a validation exception.